src/engine/enginebuffer.cpp

```diff
@@ -1120,7 +1121,26 @@ void EngineBuffer::processSlip(int iBufferSize) {
 
     // Increment slip position even if it was just toggled -- this ensures the position is correct.
     if (enabled) {
-        m_dSlipPosition += static_cast<double>(iBufferSize) * m_dSlipRate;
+        double slipDelta = static_cast<double>(iBufferSize) * m_dSlipRate;
+        // a regular loop is active
+        if (m_pLoopingControl->isLoopingEnabled() &&
+                !m_pLoopingControl->isLoopRollActive()) {
+            double newPos = m_dSlipPosition + slipDelta;
+            // simulate looping
+            const double loopStart = m_pLoopingControl->getLoopSamples().start;
+            const double loopEnd = m_pLoopingControl->getLoopSamples().end;
+            while (newPos > loopEnd) {
+                // passed loop_out or loop was moved backwards
+                newPos -= loopEnd - loopStart;
+            }
+            while (newPos < loopStart) {
```

Comment:
```suggestion
            while (newPos <= loopStart) {
```

Reply:
Not sure if this is can be really a long loop. But a using fmod() should be faster in any case.

Reply:
Not sure if this is can be really a long loop. But a using fmod() should be faster in any case.